plugins/module_utils/azure_rm_common.py

```diff
@@ -521,8 +521,13 @@ def has_tags(self, obj_tags, tag_list):
         for tag in tag_list:
             tag_key = tag
             tag_value = None
-            if ':' in tag:
-                tag_key, tag_value = tag.split(':')
```

Comment:
@Maarc-D In addition, if such a problem is to be solved, I suggest changing it as follows:
```e.g.
tag_key, tag_value = tag.rsplit(':',1)

Reply:
@haiyuazhang What do you think?

Reply:
That is my recommendation too. I was working on it and just before PR'ing I realize this was already thrown:

```shell
diff --git a/plugins/module_utils/azure_rm_common.py b/plugins/module_utils/azure_rm_common.py
index 62d38e2..7e02dc6 100644
--- a/plugins/module_utils/azure_rm_common.py
+++ b/plugins/module_utils/azure_rm_common.py
-                tag_key, tag_value = tag.rsplit(':', 1)
+                tag_key, tag_value = tag.split(':')

```

Reply:
@Maarc-D What do you think of the changes in the comments above?

Reply:
Hello,
The solution you proposed dont fix the issue.
Indeed no more error will be raised but the comportement will not be good.

take exemple with tag as key : "company:group:function" and value : "mybestvaluerepresentation"

if we implement as you suggested we will use : "company:group:function:mybestvaluerepresentation" and the result will be 

tag_key = company
tag_value = group:function:mybestvaluerepresentation

So it will never match.

Regards Marc.